### ISSUE TITLE
Query builder adds extra space if `like` statement has `and` or `or`

### ISSUE THREAD

**Issue by ksimka** [bcit-ci/CodeIgniter#4551]
Initially from http://stackoverflow.com/questions/36131222/getting-extra-space-in-codeigniter-like-query

Code

```php
$this->db->like('postcoderegion','Brighton And Hove');
$query = $this->db->get('postcode');
$result = $query->result();
```

produces

```sql
SELECT * FROM `postcode` WHERE  postcoderegion  LIKE '%Brighton And  Hove%' ESCAPE '!' 
```

Looks like the problem is in https://github.com/bcit-ci/CodeIgniter/blob/3.0.5/system/database/DB_query_builder.php#L2378

```php
$conditions = preg_split(
    '/((?:^|\s+)AND\s+|(?:^|\s+)OR\s+)/i',
    $this->{$qb_key}[$i]['condition'],
    -1,
    PREG_SPLIT_DELIM_CAPTURE | PREG_SPLIT_NO_EMPTY
);
```

So if `and` or `or` is a part of condition, extra space is added later with concatenation. I can be wrong, but it's worth checking.

**Comment by narfbg** [comment]
If it's worth checking, why didn't you check it?

The OP for that SO question is using version 3.0.1.
This issue was fixed in version 3.0.2: #4093

It's listed in the changelog and not reproducible with 3.0.5.

**Comment by ksimka** [comment]
Thanks for the kind response. Next time I'll better pass by.
